src/components/sidenav/sidenav.js

```diff
@@ -323,6 +326,106 @@ function SidenavDirective($mdMedia, $mdUtil, $mdConstant, $mdTheming, $animate,
       }
     }
 
+    function enableDragging() {
+      $mdGesture.register(element, 'drag', { horizontal: true });
+
+      element
+        .on('$md.dragstart', onDragStart)
+        .on('$md.drag', onDrag)
+        .on('$md.dragend', onDragEnd);
+
+      var style = getComputedStyle(element[0]);
+      var sidenavWidth = parseInt(style.width);
+      var isRightSidenav = element.hasClass('md-sidenav-right');
+      var accelerationBound = 6;
+
+      var dragCancelled = false;
+      var dragPercentage;
+      var lastOpenState;
+      var lastDistance = 0;
+      var isQuickDrag = false;
+
+      function onDragStart() {
+        if (element.hasClass('md-locked-open')) {
+          dragCancelled = true;
+        } else {
+          lastOpenState = scope.isOpen;
+          element.css($mdConstant.CSS.TRANSITION_DURATION, '0ms');
+        }
+      }
+
+      function onDrag(ev) {
+        if (dragCancelled) return;
+
+        if (!isQuickDrag) {
+          var distance = lastDistance - ev.pointer.distanceX;
+          if (isRightSidenav) {
+            isQuickDrag = distance <= -accelerationBound;
+          } else {
+            isQuickDrag = distance >= accelerationBound;
+          }
+        } else {
+          if (isRightSidenav && lastDistance > ev.pointer.distanceX ||
+              !isRightSidenav && lastDistance < ev.pointer.distanceX) {
```

Comment:
Nothing, that's by intention, because we should not interrupt the `quickDrag` if we don't drag back.